packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts

```diff
@@ -195,6 +216,11 @@ export class DynamicRemotePlugin implements WebpackPluginInstance {
     const moduleFederationLibraryType = moduleFederationSettings.libraryType ?? 'jsonp';
     const moduleFederationSharedScope = moduleFederationSettings.sharedScopeName ?? 'default';
 
+    const {
+      ModuleFederationPlugin = container.ModuleFederationPlugin,
+      ContainerPlugin = container.ContainerPlugin,
+    } = moduleFederationSettings.pluginOverride ?? {};
```

Comment:
@vojtechszocs thoughts about writing this as:
```js
const {
  ModuleFederationPlugin,
  ContainerPlugin
} = moduleFederationSettings.pluginOverride || container;
```
?

Reply:
@florkbr I've checked the above code - it would result in both `ModuleFederationPlugin` and `ContainerPlugin` variables being possibly `undefined` so we'd still need the default assignments within that destructure statement, which then beats the `|| container` part :smiley: 

The reason for destructuring `moduleFederationSettings.pluginOverride ?? {}` instead of just `moduleFederationSettings.pluginOverride` is that the `pluginOverride` object itself can be `undefined` - TypeScript strict mode detects this as well, raising type errors if the `?? {}` part is missing.

TL;DR - I'd keep the original code.

JS destructuring assignment works with a single object getting picked from - imagine if `ModuleFederationPlugin` option was defined within the `pluginOverride` object but `ContainerPlugin` was not. In such case, the `|| container` part would have no effect.

Reply:
Chatted with @vojtechszocs offline. My main concern here was readability - though we do have some limitations with object destructuring in JS. The alternative approach is not much clearer than what we have here - so I'm fine keeping as is for now.